### DISCUSSION TITLE
Will IObservableValue.observe_ and IComputedValue.observe_ be removed from exported type definitions?

### DISCUSSION THREAD

**Discussion by emereum** [mobxjs/mobx#3606] [General] [answered]
The observe and intercept methods are no longer exposed on arrays, maps, and boxes (as per [6.0.0 breaking changes](https://github.com/mobxjs/mobx/blob/main/packages/mobx/CHANGELOG.md#changes-that-might-affect-you)) but they are still present on `IObservableValue` and `IComputedValue`.

Are these methods deprecated and scheduled for removal from the exported type definitions of a later MobX release?

For context, I ask because the event handler type discriminators in these types prevent us from assigning an `IObservableValue` to an `IComputedValue` in MobX 6, which was possible in MobX 5, and removing `observe_` would resolve this.

**Comment by mweststrate** [comment, marked as answer]
At first glance that seems to be an oversight indeed and I think they can be safely removed. Feel free to submit a PR. If that causes compile errors that is probably because of internal usage, and we could solve that by a separate interface and upcast.

**Reply by emereum** [reply]
PR submitted #3608
